### PR TITLE
Fix BUG-303: preserve still-running start keys

### PR DESCRIPTION
**Pull request by The-Obstacle-Is-The-Way**
## Problem

BUG-300 preserves a Start key inside the immediate `ConcurrentRequestInProgress` refresh, but that uncertainty was local to one helper call. BUG-299's later recovery-panel abandon captured only the unchanged key. A refreshed per-user session can belong to another tab, so successfully abandoning it could rotate the only replay handle for a different same-key request that was still able to create a session.

The final review exposed the sibling transport arm: every accepted Start is indeterminate until a same-key result is consumed. Marking uncertainty only after a typed concurrent result left a thrown/timeout request vulnerable to the same later-abandon rotation.

## Solution

- Persist per-key execution uncertainty at `usePracticeSessionStart`, the lifecycle owner.
- Claim the current key's settled version and mark it uncertain before awaiting the controller.
- Treat only a returned, version-matched same-key result as a settlement observation: typed `concurrent_request_in_progress` preserves uncertainty; every other returned result clears it. Thrown/timeout outcomes publish no settlement and therefore remain fenced.
- Fence observations by key and settled version. A stale render handler is rejected before it can call the controller or mutate the newer request's UI state, and a delayed concurrent observation cannot overwrite a later settled observation.
- Extend `captureIdempotencyKeyRetirement` to refuse retirement while its captured key remains able to commit. The existing key-unchanged fence remains intact.
- Keep authoritative refresh responsible only for panel state. No per-user incomplete session is treated as request identity.
- Leave server/session semantics, the idempotency wrapper, and BUG-291/299/300 policies unchanged.

## TDD and proof

Initial red production-hook sequence:

1. Start returned typed `concurrent_request_in_progress` under K.
2. Refresh rendered another tab's session S.
3. Abandon S succeeded.
4. The next Start carried a fresh UUID instead of K.

Final-review red production-hook sequence:

1. Start under K timed out while an unrelated initial refresh remained pending.
2. That authoritative refresh rendered another tab's session S.
3. Abandon S succeeded.
4. The next Start carried a fresh UUID instead of K.

Green coverage now proves:

- both sequences reuse K;
- a returned same-key result clears uncertainty so ordinary retirement remains possible;
- a delayed concurrent observation cannot re-poison a key after a newer settled result;
- a stale handler for an older intent makes no controller call, leaves the newer UI state unchanged, and cannot erase uncertainty owned by the newer key;
- a thrown/timeout outcome cannot be falsely settled by later unrelated-session recovery;
- existing ordinary-abandon and second-tab proven-absence retirement controls remain green;
- BUG-300's immediate loaded-null guard remains green.

The stale-handler test was added after Codecov identified the guard as the sole uncovered new line. It is behavior-scoped rather than percentage-driven: mutating the guard back to the unsafe stale-key reset made the production-hook Chromium sequence fail; restoring the guard made it green.

A `StartPracticeSessionUseCase` orchestration test creates and ends session S during candidate selection—after the initial incomplete-session precheck and before the original `sessions.create`—then proves the original request creates distinct session T. This pins why refresh result S cannot establish request identity.

## Review record

- Initial head `f774d15a` received CodeRabbit formal APPROVED with zero review threads after the first full cooldown.
- Codecov's uncovered concurrency-guard line was independently adjudicated and addressed with a non-vacuous stale-handler behavior test on `2493dd11`.
- A fresh full review on `2493dd11`, requested only after the stated 45-minute cooldown, correctly found that claim-time uncertainty was missing for thrown/timeout outcomes. The finding was reproduced red, fixed at the lifecycle owner, documented, and covered on `bbd04d38`.
- The next exact-head full review correctly found that an obsolete render handler still entered `startSession`, submitting stale intent and potentially stranding the current UI in `loading`. The existing browser test failed red with controller call #2; `5bb9fcc9` rejects the stale owner claim before any controller or UI effect and pins current-key reuse after recovery.
- The push-triggered incremental review on `5bb9fcc9` reached the shared rate limit. The head is frozen for the full stated 45-minute cooldown.
- Merge remains blocked until a fresh CodeRabbit review gives formal APPROVED on this exact head.

## Verification

Full gate passed before each push. Latest evidence on `5bb9fcc9`:

- typecheck
- lint (pre-existing advisory warnings only)
- unit: 3,394 passed
- browser: 392 passed
- integration against the per-clone local Postgres target: 200 passed, 2 skipped
- production build
- hermetic E2E: clean rerun, 36/36 passed

## Register

- Updates the [BUG-303 SSOT](https://github.com/The-Obstacle-Is-The-Way/naltrexone-university/blob/fix/bug-303-abandon-retires-running-start-key/docs/bugs/bug-303-concurrent-start-abandon-retires-still-running-key.md) with Resolution State while leaving Status Open.
- `docs/bugs/index.md` is intentionally untouched; archival waits for production proof at the wave-5 close.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved practice-session recovery when concurrent start requests are in progress.
  * Prevented session recovery from incorrectly reusing/retiring the wrong in-progress start attempt.
  * Ensured delayed or stale observations can’t overwrite newer recovery outcomes, and added idempotency-aware uncertainty handling.
  * Improved reliability when sessions change between validation and creation.
* **Tests**
  * Added coverage for concurrent-start edge cases, stale recovery results, idempotency-key rotation, and cross-session recovery/interleaving.
* **Documentation**
  * Expanded BUG-303 resolution state and verification steps.

### CODE REVIEW

File: app/(app)/app/practice/hooks/use-practice-session-start.ts [hunk 1]
```diff
@@ -66,17 +77,64 @@ export function usePracticeSessionStart(
...
+        }
+
+        startExecutionUncertaintyRef.current = mayStillFinish
+          ? { ...current, mayStillFinish: true }
+          : {
+              ...current,
+              settledVersion: current.settledVersion + 1,
+              mayStillFinish: false,
+            };
+      };
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply app/(app)/app/practice/hooks/use-practice-session-start.ts]: Confirmed from first principles. An accepted Start was not marking the owner slot uncertain; because the thrown/timeout arm intentionally publishes no returned-result observation, a later successful abandon could retire its still-indeterminate key. I added a red production-hook Chromium sequence (timeout under K → deferred authoritative refresh renders unrelated S → successful abandon S → retry) that failed because the retry used a fresh UUID. Commit bbd04d38 now marks a current-key claim uncertain before the await and leaves the existing key/version-fenced returned observation as the only clearing path. The test is green, the SSOT wording is corrected, and the full gate is green: unit 3,394; browser 392; integration 200 (+2 skipped); build; clean hermetic E2E 36/36.

**Comment by The-Obstacle-Is-The-Way** [comment]
@coderabbitai full review

File: app/(app)/app/practice/hooks/use-practice-session-start.ts [hunk 2]
```diff
@@ -66,17 +77,68 @@ export function usePracticeSessionStart(
...
 
+  const claimStartExecutionUncertainty = useCallback(
+    (idempotencyKey: string) => {
+      if (
+        startExecutionUncertaintyRef.current.idempotencyKey !== idempotencyKey
+      ) {
+        // A stale render may still invoke an old handler after a newer intent
+        // owns the slot. It must not replace the newer key's uncertainty.
+        return () => {};
+      }
```
[the remark this answers is not in the document]
Comment (The-Obstacle-Is-The-Way) [reply app/(app)/app/practice/hooks/use-practice-session-start.ts:101]: Confirmed from first principles. The stale handler entered startSession, which set the shared status to loading before awaiting the controller; because its key was already obsolete, the later isLatestRequest fence returned without restoring status. I tightened the existing production-hook Chromium test first: it failed red because the stale click made controller call #2. Commit 5bb9fcc9 makes a stale key reject the owner claim and return before startSession. Green proof now pins zero controller calls, unchanged error status, and same-key reuse by the current handler after recovery. Full gate: 3,394 unit; 392 browser; 200 integration (+2 skipped); build; hermetic E2E 36/36.
